pkg/volume/csi/csi_plugin.go

```diff
@@ -914,3 +919,30 @@ func highestSupportedVersion(versions []string) (*utilversion.Version, error) {
 	}
 	return highestSupportedVersion, nil
 }
+
+// waitForAPIServerForever waits forever to get the APIServer Version as a proxy
+// for a healthy APIServer.
+func waitForAPIServerForever(client clientset.Interface) error {
+	const nonExistingNodeName = "nonExistingNodeName"
+	var lastErr error
+	err := wait.PollImmediateInfinite(time.Second, func() (bool, error) {
+		// Get a CSINode from API server to make sure 1) kubelet can reach API server
+		// and 2) it has enough permissions. Kubelet may have restricted permissions
+		// when it's bootstrapping TLS.
+		// https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/
+		// Using a dummy node name, IsNotFound error is expected. IsForbidden or timeout are bad.
+		_, lastErr = client.StorageV1().CSINodes().Get(context.TODO(), nonExistingNodeName, meta.GetOptions{})
+		if lastErr == nil || apierrors.IsNotFound(lastErr) {
```

Comment:
lastErr shouldn't be nil if the node is non-existent :-)

Reply:
you never know :-)

Reply:
One could crash whole cluster by creating a node with nonExistingNodeName.

Reply:
Since the node may exist, we can give it name such as possiblyExistingNodeName